Eventually logging multiple occurances of the same exception

Eventually doesn't always recognize multiple occurrences of exceptions as being the same:
```
DeliveryError: Marcel tried to Eventually see if side nav expanded is equal to <True> 37 times over 10 seconds, but got:
    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>

    AssertionError: 
Expected: <True>
     but: was <False>
```

--- a/screenpy/actions/eventually.py
+++ b/screenpy/actions/eventually.py
@@ -3,7 +3,8 @@ Eventually perform a Task or Action, trying until a set timeout.
 """
 
 import time
-from typing import Dict, Optional
+from traceback import format_tb
+from typing import Optional
 
 from screenpy.actor import Actor
 from screenpy.configuration import settings
@@ -138,7 +139,8 @@ class Eventually:
                     return
                 except Exception as exc:  # pylint: disable=broad-except
                     self.caught_error = exc
-                    self.unique_errors[exc] = None
+                    if not any(same_exception(exc, c) for c in self.unique_errors):
+                        self.unique_errors.append(exc)
 
                 count += 1
                 time.sleep(self.poll)
@@ -158,6 +160,15 @@ class Eventually:
         self.performable = performable
         self.performable_to_log = get_additive_description(self.performable)
         self.caught_error = None
-        self.unique_errors: Dict[Exception, None] = {}
+        self.unique_errors: list[BaseException] = []
         self.timeout = settings.TIMEOUT
         self.poll = settings.POLLING
+
+
+def same_exception(a: BaseException, b: BaseException) -> bool:
+    """compare exceptions to see if they match"""
+    return (
+        isinstance(a, type(b))
+        and (str(a) == str(b))
+        and (format_tb(a.__traceback__) == format_tb(b.__traceback__))
+    )

--- a/screenpy/actions/eventually.py
+++ b/screenpy/actions/eventually.py
@@ -4,7 +4,7 @@ Eventually perform a Task or Action, trying until a set timeout.
 
 import time
 from traceback import format_tb
-from typing import Optional
+from typing import List, Optional
 
 from screenpy.actor import Actor
 from screenpy.configuration import settings
@@ -160,7 +160,7 @@ class Eventually:
         self.performable = performable
         self.performable_to_log = get_additive_description(self.performable)
         self.caught_error = None
-        self.unique_errors: list[BaseException] = []
+        self.unique_errors: List[BaseException] = []
         self.timeout = settings.TIMEOUT
         self.poll = settings.POLLING
 

--- a/tests/test_actions.py
+++ b/tests/test_actions.py
@@ -300,6 +300,24 @@ class TestEventually:
             "    AssertionError: Failure #5"
         )
 
+    @mock.patch("screenpy.actions.eventually.time", autospec=True)
+    def test_mention_multiple_errors_once(self, mocked_time, Tester):
+        mocked_time.time = mock.create_autospec(time.time, side_effect=[1, 1, 1, 100])
+        mock_question = FakeQuestion()
+        mock_question.answered_by.return_value = True
+        mock_question.describe.return_value = "returns bool"
+        
+        with pytest.raises(DeliveryError) as actual_exception:
+            Eventually(See(mock_question, IsEqualTo(False))).perform_as(Tester)
+
+        assert str(actual_exception.value) == (
+            "Tester tried to Eventually see if returns bool is equal to False 3 times "
+            "over 20.0 seconds, but got:\n"
+            "    AssertionError: \n"
+            "Expected: <False>\n"
+            "     but: was <True>\n"
+        )
+
     def test_describe(self) -> None:
         mock_action = FakeAction()
         mock_action.describe.return_value = "An African or a European swallow?"

